### PR TITLE
english_2026-09: robustness fine-tune (mid-word prompt cuts, unpunctuated text) as the default English model

### PR DESCRIPTION
**Pull request by manu-lm**
Releases `english_2026-09` as the default English model and adds the training pieces it was made with.

**Models** (already on the hub, both repos)
- `english_2026-09`: the 6-layer `english_2026-04` re-distilled from a robustness-fine-tuned 24-layer teacher. Voice prompts cut in the middle of a word and text without a final punctuation mark are now in distribution. LibriSpeech test-clean, 448 prompts, CFG 1.0: intact prompts 0.91% -> 0.97% WER / UTMOS 4.364 -> 4.357 (3-seed full set: 0.87 -> 0.91%, 4.365 -> 4.358), prompts cut mid-word 1.17% -> 0.89%, unpunctuated text 0.90% -> 0.94%. A 400-vote pairwise listening test against `english_2026-04` on intact prompts came out 207-193 (tie).
- `english_2026-09_24l`: the fine-tuned teacher (sampler head frozen). Cut-prompt WER 1.25% -> 0.99%, unpunctuated 1.19% -> 1.00%, intact 1.03% -> 0.97% at UTMOS 4.28-4.30.
- `languages/english/` now carries the `english_2026-09` weights; the pre-made voice states were recomputed for all three folders.

**Training** (the released models used these with trim 0.5 s, dropout 0.3, no prompt window, lr 2e-5 / 4e-5, 30k steps on 4 GPUs)
- `data.prompt_trim_max_sec`: trims a uniform 0..N s tail from every voice prompt (on latent frames when training from precomputed latents).
- `data.final_punct_dropout`: drops the target text's final period with the given probability.
- `freeze_head`: keeps the sampler head fixed during a fine-tune, so a distilled student keeps its latent geometry and can be re-distilled from its released weights.
- `student_init_from`: warm-starts the student backbone from a checkpoint or a released `model.safetensors` (local or `hf://`); `distill_teacher_weights` accepts the same.

### CODE REVIEW

File: training/configs/robustness_finetune.yaml
Comment (manukyutai) [inline training/configs/robustness_finetune.yaml:1]: no need to release this

File: training/configs/robustness_redistill.yaml
Comment (manukyutai) [inline training/configs/robustness_redistill.yaml:1]: don't release this

File: training/eval/librispeech.py [hunk 1]
```diff
@@ -82,6 +82,8 @@ def eval_dir_name(args: argparse.Namespace, step: int) -> str:
         name += f"_n{args.num_items}"
     if args.seed:
         name += f"_seed{args.seed}"
+    if args.strip_final_punct:
```
Comment (manukyutai) [inline training/eval/librispeech.py]: no need to release that

File: training/dataloader/loader.py [hunk 2]
```diff
@@ -92,8 +97,18 @@ def _cap_prompt(self, prompt: npt.NDArray[np.float32]) -> tuple[npt.NDArray[np.f
...
+        if self.prompt_trim_max_sec > 0:
+            drop = int(self.rng.uniform(0.0, self.prompt_trim_max_sec) * self.sample_rate)
+            prompt = prompt[: max(len(prompt) - drop, min(len(prompt), self.sample_rate))]
         return prompt, len(prompt)
 
+    def _tokens(self, text: str) -> torch.Tensor:
+        """Tokenize the target text, dropping its sentence-final punctuation with
+        probability final_punct_dropout so unpunctuated prompts are in distribution."""
+        if self.final_punct_dropout > 0 and self.rng.random() < self.final_punct_dropout:
+            text = re.sub(r"[.!?;:,\u2026\s]+$", "", text) or text
```
Comment (manukyutai) [inline training/dataloader/loader.py]: Remove the final punct only if it doesn't affect how things are said, i.e. the final dot.

File: README.md [hunk 3]
```diff
@@ -53,6 +53,8 @@ uvx pocket-tts generate
 pocket-tts generate
 ```
 Modify the voice with `--voice` and the text with `--text`. We provide a small catalog of voices.
+The default English model is `english_2026-09`, a fine-tune of `english_2026-04` that stays clean when the voice prompt is cut in the middle of a word or the text has no final punctuation (see `training/configs/robustness_finetune.yaml` and `robustness_redistill.yaml`). `english_2026-04` and `english_2026-01` remain available by name.
```
Comment (manukyutai) [inline README.md]: Remove this comment.

File: uv.lock
Comment (manukyutai) [inline uv.lock:1]: Why this change?

Comment (manu-lm) [reply uv.lock:1]: Accidental: uv run re-locked with an older uv during a local test and the file got swept into the commit. Restored to main.

File: training/args.py [hunk 4]
```diff
@@ -23,6 +23,12 @@ class DataArgs:
     # so prompts vary in length and the target keeps most of the utterance.
     # <= 0 removes the window (any word boundary; full-prefix prompt).
     max_voice_prompt_sec: float = 5.0
+    # Robustness augmentations. prompt_trim_max_sec trims a uniform 0..N seconds off the
+    # end of every voice prompt (prompts cut mid-word at inference stay in distribution);
+    # final_punct_dropout drops the target text's final period with this probability
+    # (unpunctuated prompts stay in distribution). Both default off.
+    prompt_trim_max_sec: float = 0.0
```
Comment (manukyutai) [inline training/args.py]: Maybe change the default values

**Review by manukyutai — APPROVED** [review]

**Review by gabrieldemarmiesse — APPROVED** [review]
